### PR TITLE
reviewed and fixed

### PR DESCRIPTION
**Pull request by RohitRathore1**
It resolves issue #2737

### CODE REVIEW

File: docs/src/man/comparisons.md [hunk 1]
```diff
@@ -4,12 +4,30 @@ This section compares DataFrames.jl with other data manipulation frameworks in P
...
 A sample data set can be created using the following code:
 
-```julia
-using DataFrames
-using Statistics
-
-df = DataFrame(grp = repeat(1:2, 3), x = 6:-1:1, y = 4:9, z = [3:7; missing], id = 'a':'f')
-df2 = DataFrame(grp = [1, 3], w = [10, 11])
+```jldoctest
+julia> using DataFrames
```
Comment (bkamins) [inline docs/src/man/comparisons.md:8]: this is not needed.

Comment (bkamins) [reply docs/src/man/comparisons.md:8]: in this section we do not need codes to be runnable. Python/R/Stata codes are not runnable anyway. We just need to make sure that the codes are correct.

File: docs/src/man/comparisons.md [hunk 2]
```diff
@@ -57,9 +75,22 @@ In addition, when indexing a data frame with the `findfirst` function, a single
 or boolean indexing instead. It would then return a `DataFrame` object containing all matched rows. The following
 two lines of code are functionally equivalent:
 
-```julia
-df[findall(==('c'), df.id), :]
-df[df.id .== 'c', :]
+```jldoctest dataframe
```
Comment (bkamins) [inline docs/src/man/comparisons.md:78]: same here

Comment (RohitRathore1) [reply docs/src/man/comparisons.md:78]: So, we dont need to turn on doctests?

File: docs/src/man/comparisons.md [hunk 3]
```diff
@@ -100,6 +131,10 @@ examples above do not synchronize the column names between pandas and DataFrames
 
 ### Mutating operations
 
+!!! note
+    
+    Some of the operations mutate the tables so every operation assumes that it is done on the original data frame.
```
Comment (bkamins) [inline docs/src/man/comparisons.md:138]: I would add this note on the top of the whole section.

File: docs/src/man/comparisons.md [hunk 4]
```diff
@@ -120,6 +155,10 @@ over each group independently. The result of `groupby` is a `GroupedDataFrame` o
 which may be processed using the `combine`, `transform`, or `select` functions.
 The following table illustrates some common grouping and aggregation usages.
 
+!!! note
+    
+    Some of the operations mutate the tables so every operation assumes that it is done on the original data frame.
```
Comment (bkamins) [inline docs/src/man/comparisons.md:162]: no need to repeat it if you included it in the top

File: docs/src/man/comparisons.md [hunk 5]
```diff
@@ -141,14 +180,18 @@ Name: x, dtype: int64
 
 For DataFrames.jl, it looks like this:
 
-```julia
+```jldoctest
```
Comment (bkamins) [inline docs/src/man/comparisons.md:185]: doctest is not required here

File: docs/src/man/comparisons.md [hunk 6]
```diff
@@ -222,35 +265,42 @@ As in dplyr, some of these functions can be applied to grouped data frames, in w
...
+|                           | `summarize(df, across(starts_with("x"), mean))`           | `combine(df, names(df, r"^x") .=> mean)`                       |
+|                           | `summarize(df, across(c(x, y), list(max, min)))`          | `combine(df, ([:x, :y] .=> [maximum minimum])...)`             |
+| Multivariate function     | `mutate(df, cor(x, y))`                                   | `transform(df, [:x, :y] => cor)`                               |
+| Row-wise                  | `mutate(rowwise(df), min(x, y))`                          | `transform(df, [:x, :y] => ByRow(min))`                        |
+|                           | `mutate(rowwise(df), which.max(c_across(matches("^x"))))` | `transform(df, AsTable(r"^x") => ByRow(argmax))`               |
+| DataFrame as input        | `summarize(df, head(across(), 2))`                        | `combine(d -> first(d, 2), df)`                                |
+| DataFrame as output       | `summarize(df, tibble(value = c(min(x), max(x))))`        | `combine(df, :x => (x -> [minimum(x), maximum(x)]) => :value)` |
```
Comment (bkamins) [inline docs/src/man/comparisons.md:283]: this change is not OK - it changes the intention of the example` AsTable` should be added.

File: docs/src/man/comparisons.md (hunk 2 above)
Comment (bkamins) [reply docs/src/man/comparisons.md:78]: no - you cannot correctly test this file anyway. What we need is manually check if R/Python/Julia outputs are matching for all operations.

File: docs/src/man/comparisons.md (hunk 6 above)
Comment (RohitRathore1) [reply docs/src/man/comparisons.md:283]: Ok, I will add `AsTable`

**Comment by RohitRathore1** [comment]
Done. Required changes has been made.

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**